vavr/src/test/java/io/vavr/collection/AbstractMapTest.java

```diff
@@ -39,11 +39,15 @@
             public IterableAssert<T> isEqualTo(Object obj) {
                 @SuppressWarnings("unchecked")
                 final Iterable<T> expected = (Iterable<T>) obj;
-                final java.util.Map<T, Integer> actualMap = countMap(actual);
-                final java.util.Map<T, Integer> expectedMap = countMap(expected);
-                assertThat(actualMap.size()).isEqualTo(expectedMap.size());
-                actualMap.forEach((k, v) -> assertThat(v).isEqualTo(expectedMap.get(k)));
-                return this;
+                if (actual instanceof LinkedHashMap && obj instanceof LinkedHashMap) {
```

Comment:
Please override the `assertThat(Iterable<T> actual)` method in `LinkedHashMapTest` test instead. A `super` call does inherently rely on the type hierarchy, which is not necessary and error prone.

Instead `LinkedHashMapTest` should defined what equality means (even if it looks like duplicate code at a first glance).

```
@Override
protected <T> IterableAssert<T> assertThat(Iterable<T> actual) {
    ...
}

```